### PR TITLE
Support for multiple extensions

### PR DESCRIPTION
**Pull request by ibgreen**
E.g. Use same loader object for `gltf`/`glb` etc.

### CODE REVIEW

File: modules/core/src/lib/loader-utils/auto-detect-loader.js [hunk 1]
```diff
@@ -4,22 +4,45 @@ const EXT_PATTERN = /[^\.]+$/;
 // Search the loaders array argument for a loader that matches extension or text
 export function autoDetectLoader(url = '', text, loaders) {
   // Get extension
-  let extension = url.match(EXT_PATTERN);
-  if (extension && extension[0]) {
-    extension = extension[0].toLowerCase();
+  const match = url.match(EXT_PATTERN);
```
Comment (georgios-uber) [inline modules/core/src/lib/loader-utils/auto-detect-loader.js]: Would be nice to support `http://example.com/file.glb?parameter=1`.
You can try
```
x = new URL('http://example.com/file.glb?load=1')
x.pathname
```

File: modules/core/src/lib/loader-utils/auto-detect-loader.js [hunk 2]
```diff
@@ -4,22 +4,45 @@ const EXT_PATTERN = /[^\.]+$/;
...
+  return null;
+}
 
-    for (const loader of loaders) {
-      if (loader.extension === extension || loader.name.toLowerCase() === extension) {
+function findLoaderByExtension(loaders, extension) {
+  extension = extension.toLowerCase();
+
+  for (const loader of loaders) {
+    for (const loaderExtension of loader.extensions || []) {
```
Comment (georgios-uber) [inline modules/core/src/lib/loader-utils/auto-detect-loader.js]:
```suggestion
    (loader.extensions || [loader.extension]).forEach(loaderExtension => {
```

Comment (georgios-uber) [reply modules/core/src/lib/loader-utils/auto-detect-loader.js]: You don't need this at all if you normalizeLoader

**Comment by ibgreen** [comment]
> Would be nice to support http://example.com/file.glb?parameter=1.

Yes. Added your comment as TODO, should add some more unit tests before implementing this.

> You don't need this at all if you normalizeLoader

Yes, I guess as long as we ensure it is cheap enough to call multiple times, we can just call it everywhere before we use the loaders.

**Pull request merged**